packages/charts/src/components/axis-title.component.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { useContext } from 'react';
+
+import { ChartContext, ChartContextType } from '../contexts';
+import { Orientation, Line as LineType } from '../types';
+import { Layout } from '@keen.io/ui-core';
+
+import {
+  getTextAnchor,
+  getVerticalPosition,
+  getHorizontalPosition,
+} from './axis-title.utils';
+
+type GroupBox = {
+  x: number;
+  y: number;
+  height: number;
+};
+
+type Props = {
+  children: string;
+  orientation: Orientation;
+  line: LineType;
+  groupBox: GroupBox;
+  layout?: Layout;
+};
+
+const AXIS_TITLE_PADDING = 20;
+
+const AxisTitle = ({
+  children,
+  orientation,
+  line,
+  groupBox,
+  layout,
+}: Props) => {
+  const { theme } = useContext(ChartContext) as ChartContextType;
+
+  const getAxisTheme = () => {
```

Comment:
Please, extract it to separate function to avoid re-definition on each re-render + add unit tests for all cases from switch.